src/sync/installation.ts

```diff
@@ -252,6 +256,14 @@ async function doProcessInstallation(app, data: BackfillMessagePayload, sentry:
 			try {
 				return await processor(logger, github, gitHubInstallationClient, jiraHost, repository, cursor, perPage);
 			} catch (err) {
+				// In the event that the customer has not accepted the required permissions.
+				// We will continue to process the data per usual while omitting the tasks the app does not have access too.
+				if (err.status === 403 && task === "build" && err.message?.includes("Resource not accessible by integration")) {
+					await subscription?.update({ syncWarning: "Invalid permissions for build task" });
+					logger.error({ err }, `Invalid permissions for build task`);
+					// Return undefined objects so the sync can complete while skipping this task
+					return 	{ edges: undefined, jiraPayload: undefined };
+				}
```

Comment:
This is probably not the best way to do this.  Furthermore, the same thing can happen with any other tasks and should probably be handled.

Reply:
Yeah, not a big fan of this,
In terms of placement of the logic, originally tried to isolate it within the build function itself, but then thought a more generic approach was beneficial, but added on that task === "build" last minute :(

Keen to discuss alternatives though!

Reply:
I think leaving it there for now is okay as long as there's a TODO as we need a better way to manage GH errors globally.  I don't think we need to specify `task === 'build'` because it should be showing for any task that might not have access

Reply:
added a todo.
Dropped the `task === "build"`
updated the `invalid permissions for build task` to a template string `invalid permissions for${task} task`


Am currently updating the messaging on the UI to be generic also.

Reply:
Any failure on the permissions side will be treated the same, be it builds, pulls, commits etc